File: v1-cpp-cli/src/PasswordGenerator.cpp
#include "PasswordGenerator.h"

void PasswordGenerator::handlePasswordGenerator(AppState& appState) {
	int input;
	std::string generatedPassword;
	std::cin.ignore(10000, '\n');

	std::cout << "\n Password-Generator" << std::endl;
	std::cout << "====Secure-Vault====" << std::endl;
	std::cout << "[1] Generate Password" << std::endl;
	std::cout << "[0] Return to menu\n" << std::endl;

	if (!(std::cin >> input)) {
		std::cin.clear();
		std::cin.ignore(10000, '\n');
		std::cout << "Invalid input.\n" << std::endl;
		return;
	}

	switch (input) {
		case 0:
			appState = AppState::MAIN_MENU;
			return;
		case 1:
			int length;
			std::cout << "Enter desired password length: ";

			if (!(std::cin >> length)) {}
			else if (length < 6) {
				std::cout << "Length must be at least 6.\n" << std::endl;
				return;
			}

			generatedPassword = passwordGenerator(length);
			std::cout << "Generated Password: " << generatedPassword << "\n" << std::endl;
			return;
	}
}

std::string PasswordGenerator::passwordGenerator(int length) {
	std::string password;
	const std::string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()";
	std::random_device rd;
	std::mt19937 generator(rd());
	std::uniform_int_distribution<> distribution(0, (int)(chars.size() - 1));

	for (int i = 0; i < length; ++i) {
		password += chars[distribution(generator)];
	}

	return password;
}

File: v2-csharp-cli/src/PasswordGenerator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SecureVault
{
    public class PasswordGenerator
    {
        private string passwordGenerator(int length)
        {
            string password = "";
            const string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()";

            Random random = new Random();
            for (int i = 0; i < length; i++)
            {
                int index = random.Next(0, chars.Length);
                password += chars[index];
            }

            return password;
        }

        public void handlePasswordGenerator()
        {
            int input = -1;

            Console.WriteLine("\n Password-Generator");
            Console.WriteLine("====Secure-Vault====");
            Console.WriteLine("[1] Generate Password");
            Console.WriteLine("[0] Return to menu\n");

            switch (input)
            {
                case 0:
                    App.state = AppState.MAIN_MENU;
                    break;
                case 1:
                    int length = 0;
                    string password;

                    Console.WriteLine("Enter desired password length: ");
                    try
                    {
                        length = int.Parse(Console.ReadLine());
                    }
                    catch (Exception) { Console.WriteLine("You can only enter a number.\n"); }

                    password = passwordGenerator(length);
                    Console.WriteLine($"Generated password: {password}\n");
                    break;
            }
        }
    }
}
